plugins/octant/cmd/antrea-octant-plugin/traceflow.go

```diff
@@ -383,6 +334,273 @@ func (p *antreaOctantPlugin) actionHandler(request *service.ActionRequest) error
+		if err != nil {
+			return crdv1alpha1.Destination{}, err
+		}
+		err = validateNamespace(request, dstNamespace, "destination")
+		if err != nil {
+			return crdv1alpha1.Destination{}, err
+		}
+		destination = crdv1alpha1.Destination{
+			Namespace: dstNamespace,
+			Pod:       dst,
+		}
+	case crdv1alpha1.DstTypeIPv4:
+		if isLiveTraffic && !isSrcPodType {
+			alertPrinter(request, invalidInputMsg+"one of source/destination must be a Pod: "+dst,
+				"one of source/destination must be a Pod, please check your input and submit again.", nil, nil)
+			return crdv1alpha1.Destination{}, errors.New("one of source/destination must be a Pod")
+		}
+		err := validateIP(request, dst, "destination")
+		if err != nil {
+			return crdv1alpha1.Destination{}, err
+		}
+		destination = crdv1alpha1.Destination{
+			IP: dst,
+		}
+	case crdv1alpha1.DstTypeService:
+		if isLiveTraffic && !isSrcPodType {
+			alertPrinter(request, invalidInputMsg+"one of source/destination must be a Pod: "+dst,
+				"one of source/destination must be a Pod, please check your input and submit again.", nil, nil)
+			return crdv1alpha1.Destination{}, errors.New("one of source/destination must be a Pod")
+		}
+		err := validateNamespace(request, dstNamespace, "destination")
+		if err != nil {
+			return crdv1alpha1.Destination{}, err
+		}
+		if errs := validation.NameIsDNS1035Label(dst, false); len(errs) != 0 {
+			alertPrinter(request, invalidInputMsg+"failed to validate destination service string: "+dst,
+				"Invalid destination service string, please check your input and submit again.", errs, nil)
+			return crdv1alpha1.Destination{}, errors.New("invalid destination")
+		}
+		destination = crdv1alpha1.Destination{
+			Namespace: dstNamespace,
+			Service:   dst,
+		}
+	}
+	return destination, nil
+
+}
+
+func checkPorts(request *service.ActionRequest) (bool, bool, uint16, uint16) {
+	hasSrcPort, hasDstPort := true, true
+	srcPort, err := request.Payload.Uint16(srcPortCol)
+	if err != nil {
+		hasSrcPort = false
+	}
+	dstPort, err := request.Payload.Uint16(dstPortCol)
+	if err != nil {
+		hasDstPort = false
+	}
+	return hasSrcPort, hasDstPort, srcPort, dstPort
+}
+
+func checkProtocol(request *service.ActionRequest) (string, error) {
+	protocol, err := request.Payload.StringSlice(protocolCol)
+	if err != nil || len(protocol) == 0 {
+		alertPrinter(request, invalidInputMsg+"failed to get protocol as string slice: ",
+			"Failed to get protocol as string, please check your input and submit again.", nil, err)
+		return "", err
+	}
+	return protocol[0], nil
+}
+
+func checkTimeout(request *service.ActionRequest) (bool, uint16) {
+	hasTimeout := true
+	timeout, err := request.Payload.Uint16(timeoutCol)
+	if err != nil {
+		hasTimeout = false
+	}
+	return hasTimeout, timeout
+}
+
+func updateIPHeader(tf *crdv1alpha1.Traceflow, hasSrcPort bool, hasDstPort bool, srcPort uint16, dstPort uint16) {
+	switch tf.Spec.Packet.IPHeader.Protocol {
+	case crdv1alpha1.TCPProtocol:
+		{
```

Comment:
not common to wrap the code with an extra brace, which causes unnecessary indent.